omar_qaterge.js

```diff
@@ -0,0 +1,202 @@
+} else {
+    highest_value = a;
+    console.log("the highest value of the two is: " + highest_value);
+}
+
+
+
+
+
+//mothod 3
+let z = 7.25;
+console.log(z);
+const a = Math.round(z);
+console.log(a);
+const s = z - a;
+let highest_value;
+if (s > a - z) {
+	highest_value = a + s;
+    console.log("the highest value of the two is: " + highest_value);
+} else {
+	highest_value = z + -s;
+    console.log("the highest value of the two is: " + highest_value);
+}
+
+
+
+
+//method 4
+let z = 7.25;
+console.log(z);
+let a = z;
+if (a - Math.trunc(a) <= 0.4) {
+	a = Math.trunc(a);
+} else {
+	a = 1 - (a - Math.trunc(a)) + a;
+}
+console.log(a);
+let highest_value;
+if (z > a) { 
+	highest_value = z;
+    console.log("the highest value of the two is: " + highest_value);
+} else {
+    highest_value = a;
+    console.log("the highest value of the two is: " + highest_value);
+}
+
+
+
+
+//method 5
+let z = 7.25;
+console.log(z);
+let a = z;
+n = (a + "").split(".");
+i =(n.splice(-1, 1));
+let decimal = i[0];
+if (decimal.charAt(0) <= 4) {
+	a = Math.trunc(a);
+} else {
+	a = 1 - (a - Math.trunc(a)) + a;
+}
+console.log(a);
+let highest_value;
+if (z > a) { 
+	highest_value = z;
+    console.log("the highest value of the two is: " + highest_value);
+} else {
+    highest_value = a;
+    console.log("the highest value of the two is: " + highest_value);
+}
+
+
+
+
+//6
+const class_names = [];
+console.log("the value of the array is null which is nothing because it's empty");
+console.log(class_names);
+const domesticated_animals = ['dog', 'cat', 'horse', 'yak', 'alpaca'];
+console.log(domesticated_animals);
+domesticated_animals.push('baby pig');
+console.log(domesticated_animals);
+
+
+
+
+//7
+//more strings
+let myString = "this is a test";
+console.log(myString);
+console.log('the length of my string is: '+ myString.length);
+
+
+
+
+//8
+const string = 'hello, world!';
+const number = 5;
+const boolean = true;
+const object = {firstName:"Elon", lastName:"Musk"};
+console.log(string);
+console.log(number);
+console.log(boolean);
+console.log(object);
+console.log('the type of hello world is string');
+console.log(typeof string);
+console.log('the type of 5 is number');
+console.log(typeof number);
+console.log('the type of true is boolean');
+console.log(typeof boolean);
+console.log('the type of firstName:"Elon", lastName:"Musk" is object');
+console.log(typeof object);
+
+if (typeof boolean == typeof object) {
+    console.log('SAME TYPE');
+} else {
+	console.log('NOT THE SAME TYPE');
+}
+
+
+
+
+//9
+let x = 7;
+console.log("the value of x will be 1 because it's the remainder");
+console.log(x = x % 3);
+console.log('% is the modulus operator, it will let you have the remainder of the division');
+console.log('The % symbol is used in most programming languages, including JavaScript, as Modulu');
+console.log('modulo is the operation use to find the remainder after division of one number by another');
```

Comment:
👍 for the explanation but you should try avoid doing things like these: `console.log(x = x % 3);` because you are trying to cover two operations in a single statement which isn't a nice way to do things. It is much better if you separate them in separate statements.